kolibri/plugins/learn/assets/src/views/content-cards/content-carousel.vue

```diff
@@ -0,0 +1,205 @@
+    components: {
+      'icon-button': require('kolibri.coreVue.components.iconButton'),
+      'content-card': require('./content-card'),
+    },
+    data() {
+      return {
+        currentSetIndex: 0,
+        animation: 'next',
+      };
+    },
+    computed: {
+      contentSets() {
+        const contentSetsArray = chunk(this.contents, this.contentSetSize);
+        if (this.currentSetIndex >= contentSetsArray.length) {
+          this.currentSetIndex = contentSetsArray.length - 1;
+        }
+        return contentSetsArray;
+      },
+      isFirstSet() {
+        return this.currentSetIndex === 0;
+      },
+      isLastSet() {
+        return this.currentSetIndex === (this.contentSets.length - 1);
+      },
+      contentSetSize() {
+        // we can calculate these based off of the size of the cards later
+        switch (this.windowSize.breakpoint) {
+          case 0:
+            return 1;
+          case 1:
+            return 2;
+          case 2:
+            return 2;
+          case 3:
+            return 3;
+          case 4:
+            return 4;
+          case 5:
+            return 4;
+          case 6:
+            return 5;
+          default:
+            return 6;
+        }
+      },
+      widthOfCarousel() {
+        const cardWidth = 210;
+        const cardMargin = 10;
+
+        const cardGutterWidth = cardMargin * 2;
+        const allGuttersWidth = this.contentSetSize * cardGutterWidth;
+        const allCardsWidth = this.contentSetSize * cardWidth;
+
+        return {
+          width: `${allGuttersWidth + allCardsWidth}px`,
+        };
+      },
+    },
+    methods: {
+      genContentLink(id) {
+        return {
+          name: PageNames.LEARN_CONTENT,
+          params: { channel_id: this.channelId, id },
+        };
+      },
+      nextSet() {
+        this.currentSetIndex += 1;
+        this.animation = 'next';
+      },
+      previousSet() {
+        this.currentSetIndex -= 1;
+        this.animation = 'previous';
+      },
+    },
+    vuex: {
+      getters: {
+        channelId: (state) => state.core.channels.currentId,
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="stylus" scoped>
+
+  @require '~kolibri.styles.definitions'
+
+  $card-gutter = 10px
+  $card-height = 210px
+
+  .content-carousel
+    // position: absolute
+    &-header
+      text-align: left
+      margin-bottom: 1em
+      h2
+        margin: 0
+    &-controls
+      position: relative
+      top: -($card-height / 2)
+      clearfix()
+      .next, .previous
+        position: relative
+      .next
+        float: right
+      .previous
+        float: left
+
+    &-cards
+      margin-left: auto
+      margin-right: auto
+
+  .content-card
+    margin-right: $card-gutter
+    margin-left: $card-gutter
+
+  // Applies to both 'next' animation and previous' animation
+  .next, .previous
+    // setting the animation for seamless movements
+    &-enter-active, &-leave-active
+      transition: all 0.5s linear
```

Comment:
Tried a couple different kinds. Linear looked smoothest imo, but I can show you others later.